x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTestHelper.java

```diff
@@ -251,30 +253,30 @@ public static RemoteAccessAuthentication randomRemoteAccessAuthentication(RoleDe
     }
 
     public static RemoteAccessAuthentication randomRemoteAccessAuthentication() {
-        return randomRemoteAccessAuthentication(
-            new RoleDescriptorsIntersection(
-                List.of(
-                    // TODO randomize to add a second set once we have querying-cluster-side API key support
-                    Set.of(
-                        new RoleDescriptor(
-                            "_remote_user",
-                            null,
-                            new RoleDescriptor.IndicesPrivileges[] {
-                                RoleDescriptor.IndicesPrivileges.builder()
-                                    .indices("index1")
-                                    .privileges("read", "read_cross_cluster")
-                                    .build() },
-                            null,
-                            null,
-                            null,
-                            null,
-                            null,
-                            null
-                        )
+        // The common use case would be 1 (user or API key auth) or 2 (API key auth) role descriptors.
+        return randomRemoteAccessAuthentication(ESTestCase.randomIntBetween(1, 2));
+    }
+
+    public static RemoteAccessAuthentication randomRemoteAccessAuthentication(int numberOfRandomRoleDescriptors) {
+        final List<Set<RoleDescriptor>> roleDescriptors = new ArrayList<>(numberOfRandomRoleDescriptors);
+        for (int i = 0; i < numberOfRandomRoleDescriptors; i++) {
+            roleDescriptors.add(
+                Set.of(
+                    new RoleDescriptor(
+                        "_remote_user",
+                        null,
+                        RoleDescriptorTests.randomIndicesPrivileges(1, 3),
```

Comment:
> could you validate that there aren't existing tests that make an assumption about the underlying RDs? I don't think there are but to avoid random test failures, would be great to check.

I checked this before I made the change, hence why I randomized it. I'm okay with keeping the fixed privileges. We can always change it later if needed.